### ISSUE TITLE
mutt slow switching between folders with sidebar 

### ISSUE THREAD

**Issue by cycomanic** [neomutt/neomutt#186]
Hi,

mutt seems to be very slow when switching folders and using the sidebar. This seems to be scaling by the number of mailboxes. Every time I switch folders mutt displays the "Sorting Mailbox" message at the bottom for some seconds, it does not matter if the mailbox I'm switching to has 1 or 100 messages. This is using imap.

**Comment by riesebie** [comment]
* cycomanic <notifications@github.com> [2016-10-07 07:54 -0700]:

> Hi,
> 

> mutt seems to be very when switching folders and using the
> sidebar. This seems to be scaling by the number of mailboxes.
> Every time I switch folders mutt displays the "Sorting Mailbox"
> message at the bottom for some seconds, it does not matter if the
> mailbox I'm switching to has 1 or 100 messages. This is using
> imap. 

Are you using NeoMutt's header cache? What is the output of:
$ mutt -D | grep header_cache
?

Else which NeoMutt version do you run? Self compiled or distributed?

I am running mailboxes with up to 70k mails which open while just
a blob ;-)

Elimar
-- 
  Obviously the human brain works like a computer.
  Since there are no stupid computers humans can't be stupid.
  There are just a few running with Windows or even CE ;-)

**Comment by cycomanic** [comment]
Yes I'm using header_caches (LMDB backend). I've just checked at home (where my internet connection is not as fast as at work) and it is significantly slower, so this definitely seems to be an issue with neomutt making connections. This is a self compiled git checkout from a week ago or so.
I can provide my configuration if that helps.

**Comment by riesebie** [comment]
* cycomanic <notifications@github.com> [2016-10-08 11:47 -0700]:

> Yes I'm using header_caches (LMDB backend). I've just checked at home (where my internet connection is not as fast as at work) and it is significantly slower, so this definitely seems to be an issue with neomutt making connections. This is a self compiled git checkout from a week ago or so.
> I can provide my configuration if that helps.

Please provide the output of:
$ mutt -v
$ mutt --vv

Thanks
Elimar
-- 
  Do you smell something burning or is it me?

**Comment by cycomanic** [comment]
mutt -v

```
NeoMutt 20161003 (1.7.0)
Copyright (C) 1996-2016 Michael R. Elkins and others.
Mutt comes with ABSOLUTELY NO WARRANTY; for details type `mutt -vv'.
Mutt is free software, and you are welcome to redistribute it
under certain conditions; type `mutt -vv' for details.

System: Linux 4.8.0-17-generic (x86_64)
libidn: 1.33 (compiled with 1.33)
hcache backend: lmdb LMDB 0.9.18: (February 5, 2016)

Compiler:
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/6/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 6.2.0-5ubuntu11' --with-bugurl=file:///usr/share/doc/gcc-6/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-6 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-6-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-6-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-6-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 6.2.0 20160927 (Ubuntu 6.2.0-5ubuntu11) 

Configure options: '--prefix=/home/jschrod/.local/' '--enable-hcache' '--enable-notmuch' '--enable-sidebar' '--enable-compressed' '--enable-imap' '--enable-smtp' '--enable-gpgme' '--with-ssl' '--disable-external-dotlock' '--with-gnutls' '-with-sasl'

Compilation CFLAGS: -Wall -pedantic -Wno-long-long -g -O2 -fno-delete-null-pointer-checks

Compile options:
+CRYPT_BACKEND_CLASSIC_PGP +CRYPT_BACKEND_CLASSIC_SMIME +CRYPT_BACKEND_GPGME 
-DEBUG +DL_STANDALONE +ENABLE_NLS -EXACT_ADDRESS -HOMESPOOL -LOCALES_HACK 
-SUN_ATTACHMENT +HAVE_BKGDSET +HAVE_COLOR +HAVE_CURS_SET +HAVE_GETADDRINFO 
+HAVE_GETSID +HAVE_ICONV +HAVE_LANGINFO_CODESET +HAVE_LANGINFO_YESEXPR 
+HAVE_LIBIDN +HAVE_META +HAVE_REGCOMP +HAVE_RESIZETERM +HAVE_START_COLOR 
+HAVE_TYPEAHEAD +HAVE_WC_FUNCS +ICONV_NONTRANS +USE_COMPRESSED +USE_DOTLOCK 
+USE_FCNTL -USE_FLOCK -USE_FMEMOPEN -USE_GNU_REGEX -USE_GSS +USE_HCACHE 
+USE_IMAP +USE_NOTMUCH -USE_NNTP -USE_POP +USE_SASL +USE_SETGID +USE_SIDEBAR 
+USE_SMTP -USE_SSL_GNUTLS +USE_SSL_OPENSSL 
-DOMAIN
-MIXMASTER
-ISPELL
SENDMAIL="/usr/sbin/sendmail"
MAILPATH="/var/mail"
PKGDATADIR="/home/jschrod/.local/share/mutt"
SYSCONFDIR="/home/jschrod/.local/etc"
EXECSHELL="/bin/sh"

patch-attach-headers-color-neomutt
patch-compose-to-sender-neomutt
patch-compress-neomutt
patch-cond-date-neomutt
patch-encrypt-to-self-neomutt
patch-fmemopen-neomutt
patch-forgotten-attachments-neomutt
patch-ifdef-neomutt
patch-index-color-neomutt
patch-initials-neomutt
patch-keywords-neomutt
patch-kyoto-neomutt
patch-limit-current-thread-neomutt
patch-lmdb-neomutt
patch-multiple-fcc-neomutt
patch-nested-if-neomutt
patch-new-mail-neomutt
patch-nntp-neomutt
patch-notmuch-neomutt
patch-progress-neomutt
patch-quasi-delete-neomutt
patch-reply-with-xorig-neomutt
patch-sensible-browser-neomutt
patch-sidebar-neomutt
patch-skip-quoted-neomutt
patch-smime-encrypt-self-neomutt
patch-status-color-neomutt
patch-timeout-neomutt
patch-tls-sni-neomutt

To learn more about NeoMutt, visit: http://www.neomutt.org/
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

mutt -vv
```
NeoMutt 20161003 (1.7.0)
Copyright (C) 1996-2016 Michael R. Elkins <me@mutt.org>
Copyright (C) 1996-2002 Brandon Long <blong@fiction.net>
Copyright (C) 1997-2009 Thomas Roessler <roessler@does-not-exist.org>
Copyright (C) 1998-2005 Werner Koch <wk@isil.d.shuttle.de>
Copyright (C) 1999-2014 Brendan Cully <brendan@kublai.com>
Copyright (C) 1999-2002 Tommi Komulainen <Tommi.Komulainen@iki.fi>
Copyright (C) 2000-2004 Edmund Grimley Evans <edmundo@rano.org>
Copyright (C) 2006-2009 Rocco Rutte <pdmef@gmx.net>
Copyright (C) 2014-2016 Kevin J. McCarthy <kevin@8t8.us>

Many others not mentioned here contributed code, fixes,
and suggestions.

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.

To learn more about NeoMutt, visit: http://www.neomutt.org/
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
 ```

**Comment by cycomanic** [comment]
mutt -v

```
NeoMutt 20161003 (1.7.0)
Copyright (C) 1996-2016 Michael R. Elkins and others.
Mutt comes with ABSOLUTELY NO WARRANTY; for details type `mutt -vv'.
Mutt is free software, and you are welcome to redistribute it
under certain conditions; type `mutt -vv' for details.

System: Linux 4.8.0-17-generic (x86_64)
libidn: 1.33 (compiled with 1.33)
hcache backend: lmdb LMDB 0.9.18: (February 5, 2016)

Compiler:
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/6/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 6.2.0-5ubuntu11' --with-bugurl=file:///usr/share/doc/gcc-6/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-6 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-6-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-6-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-6-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 6.2.0 20160927 (Ubuntu 6.2.0-5ubuntu11) 

Configure options: '--prefix=/home/jschrod/.local/' '--enable-hcache' '--enable-notmuch' '--enable-sidebar' '--enable-compressed' '--enable-imap' '--enable-smtp' '--enable-gpgme' '--with-ssl' '--disable-external-dotlock' '--with-gnutls' '-with-sasl'

Compilation CFLAGS: -Wall -pedantic -Wno-long-long -g -O2 -fno-delete-null-pointer-checks

Compile options:
+CRYPT_BACKEND_CLASSIC_PGP +CRYPT_BACKEND_CLASSIC_SMIME +CRYPT_BACKEND_GPGME 
-DEBUG +DL_STANDALONE +ENABLE_NLS -EXACT_ADDRESS -HOMESPOOL -LOCALES_HACK 
-SUN_ATTACHMENT +HAVE_BKGDSET +HAVE_COLOR +HAVE_CURS_SET +HAVE_GETADDRINFO 
+HAVE_GETSID +HAVE_ICONV +HAVE_LANGINFO_CODESET +HAVE_LANGINFO_YESEXPR 
+HAVE_LIBIDN +HAVE_META +HAVE_REGCOMP +HAVE_RESIZETERM +HAVE_START_COLOR 
+HAVE_TYPEAHEAD +HAVE_WC_FUNCS +ICONV_NONTRANS +USE_COMPRESSED +USE_DOTLOCK 
+USE_FCNTL -USE_FLOCK -USE_FMEMOPEN -USE_GNU_REGEX -USE_GSS +USE_HCACHE 
+USE_IMAP +USE_NOTMUCH -USE_NNTP -USE_POP +USE_SASL +USE_SETGID +USE_SIDEBAR 
+USE_SMTP -USE_SSL_GNUTLS +USE_SSL_OPENSSL 
-DOMAIN
-MIXMASTER
-ISPELL
SENDMAIL="/usr/sbin/sendmail"
MAILPATH="/var/mail"
PKGDATADIR="/home/jschrod/.local/share/mutt"
SYSCONFDIR="/home/jschrod/.local/etc"
EXECSHELL="/bin/sh"

patch-attach-headers-color-neomutt
patch-compose-to-sender-neomutt
patch-compress-neomutt
patch-cond-date-neomutt
patch-encrypt-to-self-neomutt
patch-fmemopen-neomutt
patch-forgotten-attachments-neomutt
patch-ifdef-neomutt
patch-index-color-neomutt
patch-initials-neomutt
patch-keywords-neomutt
patch-kyoto-neomutt
patch-limit-current-thread-neomutt
patch-lmdb-neomutt
patch-multiple-fcc-neomutt
patch-nested-if-neomutt
patch-new-mail-neomutt
patch-nntp-neomutt
patch-notmuch-neomutt
patch-progress-neomutt
patch-quasi-delete-neomutt
patch-reply-with-xorig-neomutt
patch-sensible-browser-neomutt
patch-sidebar-neomutt
patch-skip-quoted-neomutt
patch-smime-encrypt-self-neomutt
patch-status-color-neomutt
patch-timeout-neomutt
patch-tls-sni-neomutt

To learn more about NeoMutt, visit: http://www.neomutt.org/
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

mutt -vv
```
NeoMutt 20161003 (1.7.0)
Copyright (C) 1996-2016 Michael R. Elkins <me@mutt.org>
Copyright (C) 1996-2002 Brandon Long <blong@fiction.net>
Copyright (C) 1997-2009 Thomas Roessler <roessler@does-not-exist.org>
Copyright (C) 1998-2005 Werner Koch <wk@isil.d.shuttle.de>
Copyright (C) 1999-2014 Brendan Cully <brendan@kublai.com>
Copyright (C) 1999-2002 Tommi Komulainen <Tommi.Komulainen@iki.fi>
Copyright (C) 2000-2004 Edmund Grimley Evans <edmundo@rano.org>
Copyright (C) 2006-2009 Rocco Rutte <pdmef@gmx.net>
Copyright (C) 2014-2016 Kevin J. McCarthy <kevin@8t8.us>

Many others not mentioned here contributed code, fixes,
and suggestions.

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.

To learn more about NeoMutt, visit: http://www.neomutt.org/
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
 ```

**Comment by cycomanic** [comment]
I am accessing an exchange server through davmail, and so I could just check the davmail logs. Mutt seems to fetch every message in all folders on the server (which explains the slowness), see below for a part of the davmail log:
```
2016-10-08 20:42:08,723 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Folder Papers/Alerts - Search items current count: 5 fetchCount: 500 highest uid: 5 lowest uid: 1
2016-10-08 20:42:08,724 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 1 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAAAHzUAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlAAADAjjYd5ubGQ6z14Ti0B/5WAAAAAB81AAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAAAATq
2016-10-08 20:42:08,724 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 2 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAAB+bIAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlAAADAjjYd5ubGQ6z14Ti0B/5WAAAAAfmyAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAAAEdp
2016-10-08 20:42:08,724 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 3 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAAB+bMAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlAAADAjjYd5ubGQ6z14Ti0B/5WAAAAAfmzAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAAAEdq
2016-10-08 20:42:08,724 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 4 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAwTOrEAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlAAADAjjYd5ubGQ6z14Ti0B/5WAAAMEzqxAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAMEt03
2016-10-08 20:42:08,724 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 5 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAwTOrIAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlAAADAjjYd5ubGQ6z14Ti0B/5WAAAMEzqyAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAMEt04
2016-10-08 20:42:08,724 DEBUG [ImapConnection-54126] davmail  - >  STATUS "Papers/Alerts" (UIDNEXT 6 UIDVALIDITY 1 UNSEEN 0 RECENT 0 MESSAGES 5)
2016-10-08 20:42:08,724 DEBUG [ImapConnection-54126] davmail  - > a0080 OK STATUS completed
2016-10-08 20:42:08,726 DEBUG [ImapConnection-54126] davmail  - < a0081 STATUS "Papers/Drafts" (UIDNEXT UIDVALIDITY UNSEEN RECENT MESSAGES)
2016-10-08 20:42:09,545 DEBUG [ImapConnection-54126] davmail.imap.ImapConnection  - *
2016-10-08 20:42:10,001 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Folder Papers/Drafts - Search items current count: 2 fetchCount: 500 highest uid: 2 lowest uid: 1
2016-10-08 20:42:10,001 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 1 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAwTaasAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlIAADAjjYd5ubGQ6z14Ti0B/5WAAAME2mrAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAMExkH
2016-10-08 20:42:10,001 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 2 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAwTaawAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlIAADAjjYd5ubGQ6z14Ti0B/5WAAAME2msAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAMExkI
2016-10-08 20:42:10,001 DEBUG [ImapConnection-54126] davmail  - >  STATUS "Papers/Drafts" (UIDNEXT 3 UIDVALIDITY 1 UNSEEN 0 RECENT 0 MESSAGES 2)
2016-10-08 20:42:10,001 DEBUG [ImapConnection-54126] davmail  - > a0081 OK STATUS completed
2016-10-08 20:42:10,003 DEBUG [ImapConnection-54126] davmail  - < a0082 STATUS "Papers/Submittted" (UIDNEXT UIDVALIDITY UNSEEN RECENT MESSAGES)
2016-10-08 20:42:10,769 DEBUG [ImapConnection-54126] davmail.imap.ImapConnection  - *
2016-10-08 20:42:11,325 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Folder Papers/Submittted - Search items current count: 5 fetchCount: 500 highest uid: 5 lowest uid: 1
2016-10-08 20:42:11,325 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 1 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAAAJzUAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlHAADAjjYd5ubGQ6z14Ti0B/5WAAAAACc1AAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAAAAUV
2016-10-08 20:42:11,325 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 2 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAABTa0AAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlHAADAjjYd5ubGQ6z14Ti0B/5WAAAAAU2tAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAAAFfm
2016-10-08 20:42:11,325 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 3 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAwTZcMAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlHAADAjjYd5ubGQ6z14Ti0B/5WAAAME2XDAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAMExTn
2016-10-08 20:42:11,325 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 4 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAwTZcQAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlHAADAjjYd5ubGQ6z14Ti0B/5WAAAME2XEAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAMExTo
2016-10-08 20:42:11,325 DEBUG [ImapConnection-54126] davmail.exchange.ExchangeSession  - Message IMAP uid: 5 uid: AAAAAOYroUEY99xFk2z6mEE9XY0BAMCONh3m5sZDrPXhOLQH/lYAAAwTZcUAAA== ItemId: AAMkADc4OGUxODNhLTY1N2EtNGQzYi1iNGU2LTcyMzcxMWQ1YjkwNwBGAAAAAADmK6FBGPfcRZNs+phBPV2NBwDAjjYd5ubGQ6z14Ti0B/5WAAAAAAlHAADAjjYd5ubGQ6z14Ti0B/5WAAAME2XFAAA= ChangeKey: CQAAABYAAADAjjYd5ubGQ6z14Ti0B/5WAAAMExTp
2016-10-08 20:42:11,325 DEBUG [ImapConnection-54126] davmail  - >  STATUS "Papers/Submittted" (UIDNEXT 6 UIDVALIDITY 1 UNSEEN 0 RECENT 0 MESSAGES 5)
2016-10-08 20:42:11,326 DEBUG [ImapConnection-54126] davmail  - > a0082 OK STATUS completed
2016-10-08 20:42:11,328 DEBUG [ImapConnection-54126] davmail  - < a0083 STATUS "Parts" (UIDNEXT UIDVALIDITY UNSEEN RECENT MESSAGES)
2
```

**Comment by cycomanic** [comment]
OK this is getting weirder, I have a second account that I access via imaps and there I don't see the issue (the account is also much bigger). The settings are essentially the same, so I'm a bit at a loss. The only difference as far as I can see is that one is accessed via imaps and the other via imap through local davmail.

**Comment by cycomanic** [comment]
OK if I understand the logs correctly, it is actually davmail that fetches all emails from the server to give an answer about the mail stats? I'm not sure how it works in thunderbird (where I don't see the same delays)

**Comment by flatcap** [comment]
hi @cycomanic 
Thanks for all the info and debugging.
There are a few things I'd like you to test, if you have the time.

First, let's rule out the sidebar.
**Test**:
- Sidebar disabled in config: `set sidebar_visible = no`
- Sidebar disabled in build: `./configure` without `--enable-sidebar`

I expect these to tests to make no difference -- still slow.

**Test**:
- Disable extra mailbox checks: `set mail_check_stats = no`

Periodically, neomutt asks for detailed info about the mailboxes, e.g. Number of new emails.
This happens, every `mail_check_stats_interval` seconds.  This **can** cause a slowdown, but it shouldn't affect opening new mailboxes.

If it's still slow, then your suspicion about davmail may be right.

Neomutt only has one mailbox open at a time.  While it's open, neomutt is listening for changes to the mailbox.

When you open another mailbox, neomutt asks the server for updated info about the emails (even if you have a header cache).  It only requests the header info, so it should be fairly quick.  Afterwards, looking through the index, neomutt uses the info from the header cache.

If Thunderbird isn't slow changing between mailboxes, that you've already viewed, then it's probably keeping several mailboxes open at once.

**Comment by cycomanic** [comment]
I could have sworn that I did check the sidebar on/off, but it seems I was wrong. All settings don't make a difference except for `set imap_check_subscribed` (or the number of enabled mailboxes) (I probably did this at the same time as the sidebar, hence I associated the slowness with the sidebar). If I either `set imap_check_subscribed` or put several mailboxes "sorting mailbox"  becomes extremely slow (and the davmail logs show all mailboxes being opened). This only happens when I use davmail (normal imap seems fine), so I suspect this davmail deciding that it needs to access every mailbox. I'm just surprised that `set mail_check_stats=no` did not make a difference.

**Comment by flatcap** [comment]
@cycomanic 
I'm closing this issue because I don't think there is anything we can change in NeoMutt to help you with this problem.  If some new information arises, or inspiration strikes someone, we can reopen the issue.
FlatCap.

**Comment by arvindsv** [comment]
Just in case it helps someone, this was happening with me as well, even with notmuch. So, nothing to do with waiting for connections to any servers.

I found that there were two aspects affecting switching speed and lots of waiting in `Sorting mail ...`:

1. Using `~b` as [a pattern](https://neomutt.org/guide/advancedusage.html#patterns) in colors. I used to have lines like this:

```
color index         color208        color234        "(~C mention@noreply.github.com ~b @somename) | (~v ~(~C mention@noreply.github.com ~b @somename))"
```

I switched it to the one below and it helped a lot ("To" check first, before body check):

```
color index         color208        color234        "(~C mention@noreply.github.com ~b @somename) | (~v ~(~C mention@noreply.github.com ~b @somename))"
```

2. Using `~X` in `index_format`. That would obviously load every message and count its attachments.

```
%?X?ATT&   ?
```

I got rid of it.

All this slowness was with header and body cache turned on.
